### PR TITLE
Document Delta Lake support for GCS

### PR DESCRIPTION
**Pull request by alexjo2144**
## Description

Documentation only

## Related issues, pull requests, and links

Read supported added by https://github.com/trinodb/trino/pull/12144

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

### CODE REVIEW

**Review by ebyhr — APPROVED** [review]

**Comment by findepi** [comment]
LGTM.
@mosabua PTAl

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 1]
```diff
@@ -17,8 +17,8 @@ Requirements
 To connect to Databricks Delta Lake, you need:
 
 * Tables written by Databricks Runtime 7.3 LTS and 9.1 LTS are supported.
-* Deployments using AWS, HDFS, and Azure Storage are fully supported. Using
-  Google Cloud Storage is not supported.
+* Deployments using AWS, HDFS, and Azure Storage are fully supported. Read
+  operations against Google Cloud Storage (GCS) are also supported.
```
Comment (grantatspothero) [inline docs/src/main/sphinx/connector/delta-lake.rst:21]: Why only read operations? Is it just a lack of integration tests against GCS to give confidence?

Comment (alexjo2144) [reply docs/src/main/sphinx/connector/delta-lake.rst:21]: insert/update/delete operations will fail

Grant reminded me that CTAS and `create table` are supported, I'll clarify that

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 2]
```diff
@@ -17,13 +17,16 @@ Requirements
 To connect to Databricks Delta Lake, you need:
 
 * Tables written by Databricks Runtime 7.3 LTS and 9.1 LTS are supported.
-* Deployments using AWS, HDFS, and Azure Storage are fully supported. Using
-  Google Cloud Storage is not supported.
+* Deployments using AWS, HDFS, and Azure Storage are fully supported. Google
+  Cloud Storage (GCS) is partially supported, see below.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.rst:21]: remove the "see below" and instead make "partially supported" a link

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 3]
```diff
@@ -59,9 +62,11 @@ including the metastore :ref:`Thrift <hive-thrift-metastore>` and :ref:`Glue
...
 
 To configure access to S3 and S3-compatible storage, Azure storage, and others,
-consult the :doc:`Amazon S3 </connector/hive-s3>` section of the Hive connector
-documentation or the :doc:`Azure storage documentation </connector/hive-azure>`,
-respectively.
+consult the appropriate section of the Hive documentation.
+
+* :doc:`Amazon S3 </connector/hive-s3>`
+* :doc:`Azure storage documentation </connector/hive-azure>`
+* :ref:`GCS <hive-google-cloud-storage-configuration>`
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.rst:69]: ultimately we should consider moving all the GCS stuff into a separate page .. but that can be a separate PR

**Review by mosabua — APPROVED** [review]
